go/libkb/skb_keyring.go

```diff
@@ -312,30 +352,30 @@ func (k *SKBKeyringFile) GetFilename() string { return k.filename }
 
 // WriteTo is similar to GetFilename described just above in terms of
 // locking discipline.
-func (k *SKBKeyringFile) WriteTo(w io.Writer) (int64, error) {
+func (k *SKBKeyringFile) WriteTo(w io.Writer) (n int64, err error) {
 	k.G().Log.Debug("+ SKBKeyringFile WriteTo")
 	defer k.G().Log.Debug("- SKBKeyringFile WriteTo")
-	packets := make(KeybasePackets, len(k.Blocks))
-	var err error
+	b64 := base64.NewEncoder(base64.StdEncoding, w)
+	defer func() {
+		// explicitly check for error on Close:
+		if closeErr := b64.Close(); closeErr != nil {
+			k.G().Log.Warning("SKBKeyringFile: WriteTo b64.Close() error: %s", closeErr)
+			if err == nil {
+				n = 0
+				err = closeErr
+				return
+			}
+		}
+		k.G().Log.Debug("SKBKeyringFile: b64 stream closed successfully")
+	}()
+
 	for i, b := range k.Blocks {
-		if packets[i], err = b.ToPacket(); err != nil {
+		err = EncodePacket(b, b64)
```

Comment:
No need to convert to a list of packets first and then encode that, since w is a temp file that's erased if an error is returned